src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -209,6 +229,23 @@ public async Task<bool> TryGetAsync(ArrayBuilder<DiagnosticData> list, Cancellat
                 }
             }
 
+            private bool MatchesPriority(DiagnosticAnalyzer analyzer)
+            {
+                // If caller isn't asking for prioritized result, then run all analyzers.
+                if (_highPriority == null)
+                    return true;
+
+                // Even if the caller is asking for prioritized results, we always need to
+                // run the compiler analyzer to get the basic set of diagnostics that analyzers
+                // need to process.
+                if (analyzer.IsCompilerAnalyzer())
+                    return true;
+
+                // Otherwise, check our special internal flag to tell.
+                var analyzerIsHighPri = analyzer is IBuiltInAnalyzer { IsHighPriority: true };
+                return analyzerIsHighPri == _highPriority;
+            }
```

Comment:
@mavasani there's still something i'm uncertain about.  It appears as if diagnostics has both a push and pull mode.  I think i've update the pull side so taht if lightbulbs cause us to compute this data, that we compute the high pri items first, then the normal pri.

However, there's still the case that the incremental analyzer is running and producing (i.e pushing) results, and i'd like it to also run teh priority analyzers first in that case.

Am i widely off base here? Should that be something we want to do in this pr?  Or is it unnecessary here for reasons that i don't quite understand currently :)

Reply:
You are correct. Your current code changes are fine, but only work for the Light bulb path for fetching diagnostics for a span. I think you want to move the code filtering the analyzers based on the priority further down the callstack, i.e. pass in `highPriority` flag to [DocumentAnalysisScope ctor](https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Diagnostics/DocumentAnalysisScope.cs,b80ffac13be08af4,references). This scope will be created for both push/pull diagnostics - you just to ensure that you pass in the appropriate priority values at all construction sites for `DocumentAnalysisScope`